cluster/v1alpha1/helpers.go

```diff
@@ -181,13 +205,21 @@ func (r *RolloutHandler[T]) getProgressivePerGroupClusters(rolloutStrategy Rollo
 	if strategy.ProgressivePerGroup == nil {
 		strategy.ProgressivePerGroup = &RolloutProgressivePerGroup{}
 	}
+	minSuccessTime := rolloutStrategy.ProgressivePerGroup.MinSuccessTime.Duration
+	maxFailures := rolloutStrategy.ProgressivePerGroup.MaxFailures
 
 	// Parse timeout for non-mandatory decision groups
-	failureTimeout, err := parseTimeout(strategy.ProgressivePerGroup.Timeout.Timeout)
+	failureTimeout, err := parseTimeout(strategy.ProgressivePerGroup.ProgressDeadline)
 	if err != nil {
 		return &strategy, RolloutResult{}, err
 	}
 
+	// Check format of MaxFailures--this value will be re-parsed and used in progressivePerGroup()
+	_, err = calculateRolloutSize(maxFailures, 100, 0)
```

Comment:
Can understand the intension here. Is that possible to split the calculateRolloutSize into a parse function and a calculate function, so that don't need to hardcode a 100 which looks strange.

Reply:
Ah, good point--I'll update it.